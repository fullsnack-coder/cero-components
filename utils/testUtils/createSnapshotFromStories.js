import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
