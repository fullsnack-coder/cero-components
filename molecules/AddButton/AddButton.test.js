import * as stories from './AddButton.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ molecules / AddButton ]', () => {
  describe('#AddButton', () => createSnapshotFromStories('AddButton', stories))
})
