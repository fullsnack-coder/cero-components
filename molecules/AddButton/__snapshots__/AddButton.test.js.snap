// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ molecules / AddButton ] #AddButton when 'AddButton' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="add-button type-primary"
    id="-add-button"
  >
    <div
      class="icon color-base size-md background-highlight"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 20 20"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="spacer horizontal"
      style="height: 100%; max-height: auto; width: 15px;"
    />
    <p
      class="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </p>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / AddButton ] #AddButton when 'AddButton' EditMode is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="add-button type-primary is-editable"
    id="-add-button"
  >
    <div
      class="edit-container"
    >
      <input
        id="-add-button-input"
        type="text"
        value="Fixed caught squash wax Silvan sweet-taking keeps."
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / AddButton ] #AddButton when 'AddButton' HelpText is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="add-button type-primary is-editable"
    id="-add-button"
  >
    <div
      class="edit-container"
    >
      <input
        id="-add-button-input"
        type="text"
        value="Understand bravely halfling's rank balls."
      />
      <p
        class="help-text paragraph color-inverted size-sm weight-normal"
      >
        Morgul-rats cripples you'll?
      </p>
      <div
        class="spacer horizontal"
        style="height: 100%; max-height: auto; width: 15px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / AddButton ] #AddButton when 'AddButton' Invalid is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="add-button type-primary is-editable is-focused is-invalid"
    id="-add-button"
  >
    <div
      class="edit-container"
    >
      <input
        id="-add-button-input"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / AddButton ] #AddButton when 'AddButton' Types is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="add-button type-primary"
    id="-add-button"
  >
    <div
      class="icon color-base size-md background-highlight"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 20 20"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="spacer horizontal"
      style="height: 100%; max-height: auto; width: 15px;"
    />
    <p
      class="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </p>
  </div>
  <div
    class="add-button type-secondary"
    id="-add-button"
  >
    <div
      class="icon color-highlight size-md"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 20 20"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="spacer horizontal"
      style="height: 100%; max-height: auto; width: 15px;"
    />
    <p
      class="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </p>
  </div>
</DocumentFragment>
`;
